cstore_reader.c

```diff
@@ -858,13 +1018,13 @@ ProjectedColumnMask(uint32 columnCount, List *projectedColumnList)
 
 
 /*
- * DeserializeBoolArray reads an array of bits from the given buffer and returns
- * it as a boolean array.
+ * DeserializeBoolArray reads an array of bits from the given buffer and stores
+ * it in provided bool array.
  */
-static bool *
-DeserializeBoolArray(StringInfo boolArrayBuffer, uint32 boolArrayLength)
+static void
+DeserializeBoolArray(StringInfo boolArrayBuffer, bool *boolArray,
+					 uint32 boolArrayLength)
```

Comment:
We need a coding rule here.
I usually do following
- if integer specifically contains the length of the buffer, it comes after the buffer declaration
- if integer has more meanings like having number of items in certain collection to be put inside a provided buffer, it may come before buffer declaration.

I made quick check on citusDB codebase but did not find enough examples to reach a decision.